src/Compilers/CSharp/Portable/Symbols/UnboundGenericType.cs

```diff
@@ -119,4 +119,74 @@ public override int GetHashCode()
                 : Hash.Combine(_name, _errorInfo.Code);
         }
     }
+
+    /// <summary>
+    /// Used for lightweight binding of type constraints. Instead of binding type arguments,
+    /// we'll just use these placeholders instead. That's good enough binding to compute
+    /// <see cref="TypeSymbol.IsValueType"/> with minimal binding.
+    /// </summary>
+    internal sealed class PlaceholderTypeArgumentSymbol : ErrorTypeSymbol
+    {
+        private static readonly TypeWithAnnotations s_instance = TypeWithAnnotations.Create(new PlaceholderTypeArgumentSymbol());
```

Comment:
>private [](start = 8, length = 7)

Is there value in also having an array of one element, for the common case?
```C#
private static readonly ImmutableArray<TypeWithAnnotations> singleArgument = ImmutableArray.Create(s_instance);
```

Reply:
Should we special-case for two or three as well?
I prefer to leave as-is (ie. doesn't seem worth caching more).

Reply:
> Should we special-case for two or three as well?

If we cache any arrays, I think it's sufficient to cache the one element case. I think I'd prefer no caching over handling more cases.